### PR TITLE
Standardize DASH tests location.

### PR DESCRIPTION
**Pull request by aputriax**
- Moved all existing tests under test/test-cases
- Updated docker files to add/mount new location
- Updated documentation.

Signed-off-by: Anton Putria <antonx.putria@intel.com>

### CODE REVIEW

**Comment by aputriax** [comment]
@chrispsommers @mhanif I merged and moved your tests to test/test-cases/functional/saidashvnet_experimental.py

This branch is still WIP.

@chrispsommers I need your advise what to do with pytest folder. It seems that we need to move it to test/test-cases/functional as well. Should it be test/test-cases/pytest? Any thoughts?

File: dash-pipeline/Makefile [hunk 1]
```diff
@@ -390,23 +390,23 @@ run-saithrift-dev-pytests:deploy-ixiac disable-ipv6
 # Note - if add ixia-c usage to PTF tests, add dependency deploy-ixiac
 run-saithrift-ptftests: disable-ipv6
 	$(DOCKER_RUN_SAITHRIFT_CLIENT) \
-	-w /tests/saithrift/ptf \
+	-w /tests/ \
 	$(DOCKER_SAITHRIFT_CLIENT_IMG) \
-	./run-saithrift-ptftests.sh
+	./run-functional-tests.sh
```
Comment (chrispsommers) [inline dash-pipeline/Makefile:395]: We could continue to run tests in both locations in perpetuity. dash-pipeline tests could continue to be developed in incubation mode, let's discuss in community.

**Comment by chrispsommers** [comment]
Hi Anton, As we discussed, the idea here (which I support) is to routinely "promote" tests from dash-pipeline/tests to /tests and use this location for stable production tests, and relegate dash-pipeline/tests for incubating tests. This should be presented to the community for buy-in and education.

Regarding moving some tests to test/test-cases/functional/saidashvnet_experimental.py, it should be done in one PR, i.e. I don't like the idea of deleting them in this PR, and thereby losing the CI testing until a later PR restores them elsewhere.

File: test/test-cases/functional/saidashvnet_experimental.py [hunk 2]
```diff
@@ -0,0 +1,390 @@
+from sai_thrift.sai_headers import *
```
Comment (aputriax) [inline test/test-cases/functional/saidashvnet_experimental.py:1]: Rename the file to saidashvnet_basic.py

**Comment by chrispsommers** [comment]
@vmytnykx @aputriax Please submit a PR to https://github.com/Azure/DASH/tree/PR277-staging-branch. Make sure you've updated the dockerfile tags. I can accept the PR and it will publish the new dockerfile. Then you can submit the PR to main and it should succeed in CI because the dockerfile will be available for pulling.

**Review by chrispsommers — APPROVED** [review]

File: test/test-cases/functional/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Functional Test Cases
+Tests to verify essential functionality using low-rate traffic (SAI PTF).
+
+## Contents
+
+| Folder/File | Description |
+| --- | --- |
+| [`saidashvnet_experimental.py`](./functional/ptf/saidashvnet_experimental.py) | Basic positive VNET-to-VNET test cases designed for running on BMv2
```
Comment (chrispsommers) [inline test/test-cases/functional/README.md:8]: I think you now have saidashvnet.py and saidashvnet_sanity.py - need to update table?

File: test/test-cases/scale/README.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Scale Test Cases
+Tests with high-rate traffic and complex configuration to verify scaling real-world scenarios.
+
+## scale
+
+| Folder/File | Description |
+| --- | --- |
+| [vnet2vnet](scale/vnet2vnet/README.md) | Scale DASH vnet2vnet Tests
```
Comment (chrispsommers) [inline test/test-cases/scale/README.md:8]: link broken - remove `scale` from path

**Review by chrispsommers — CHANGES_REQUESTED** [review]
Please fix tables and links in the READMEs, otherwise LGTM.

**Review by chrispsommers — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Pull request merged**